enhancements/cluster-logging/forward_to_cloudwatch.md

```diff
@@ -0,0 +1,198 @@
+---
+title: forward_to_cloudwatch
+authors:
+  - "@alanconway"
+reviewers:
+  - "@jcantrill"
+  - "@jeremyeder"
+approvers:
+creation-date: 2020-12-17
+last-updated: 2020-12-17
+status: implementable
+see-also:
+superseded-by:
+---
+
+# Forward to Cloudwatch
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+[Amazon Cloudwatch][aws-cw] is a hosted monitoring and log storage service.
+This proposal extends the `ClusterLogForwarder` API with an output type for Cloudwatch.
+
+## Motivation
+
+Amazon Cloudwatch is a popular log store.
+We have requests from external and  Red Hat-internal customers to support it.
+
+### Goals
+
+Enable log forwarding to Cloudwatch.
+
+### Non-Goals
+
+Enable Cloudwatch metric collection.
+
+## Proposal
+
+New API fields in the `output.cloudwatch` section:
+
+- `region`: (string) AWS region name.
+- `retentionDays`: (number) Number of days to keep logs. Has a default.
+- `group`: (string, *nice-to-have*) Cloudwatch log group name.
+   Defaults to the log type (application, infrastructure or audit)
+- `groupLabel`: (string, *nice-to-have*) Kubernetes label to use as group name.
+  If the label is absent, use value of `group`.
+- `fetchIntervalSeconds`: (number, *nice-to-have*) interval to fetch logs in seconds.
+
+Existing fields:
+
+- `url`: Not used in production. Can be set for testing purposes.
+- `secret`: AWS credentials, keys `aws_access_key_id` and `aws_secret_access_key`.
+
+*Priority*: The `group`, `groupLabel` and `fetchIntervalSeconds` fields are lower priority.
+
+### User Stories
+
+#### I want to forward logs to Cloudwatch instead of a local store
+
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogForwarder"
+spec:
+  outputs:
+   - name: CloudwatchOut
+     type: cloudwatch
+     cloudwatch:
+       region: myregion
+     secret:
+        name: mysecret
+  pipelines:
+  - inputRefs: [application, infrastructure, audit]
+  - outputRefs: [CloudwatchOut]
+```
+
+[Cloudwatch][concepts] defines Log Groups and Log Streams.
+The CRD above creates 3 Log Groups named "application", "infrastructure" and "audit".
+A new Log Stream is created for each container producing logs.
+
+**Note**: The installer UI (Addon or OLM) can get AWS credentials from a `cloudcredential.openshift.io/v1`.
+The user only has to provide a `region` to enable cloudwatch forwarding for a cluster.
+Details are out of scope for this proposal.
+
+#### I want to define my own log groups
```

Comment:
Seems like a much more complicated feature to approach in the short-term.  Hopefully this won't be offered initially to reduce the complexity.

Reply:
I agree this may be a stretch goal and maybe not something ever needed, especially since we have other proposals which would allow you to "route" logs to different outputs based on a label

Reply:
+1, made that more clear.